### ISSUE TITLE
No cpanfile driver

### ISSUE THREAD

**Issue by afresh1** [shadowcat-mst/Dist-BuildProcess#3]
For things that aren't modules but do have perl/CPAN dependencies there should be support for loading the dependencies from a cpanfile while not trying to actually install the thing in the current directory.  Similar to the way `cpanm --installdeps .` works. 

This would also be useful for [perl module repositories that let Dist::Zilla create the Makefile.PL](https://github.com/afresh1/OpenSMTPd-Filter) and don't provide one otherwise.

**Comment by afresh1** [comment]
It would also be nice to have a `cpanfile.snapshot` driver that would "lock" the dependencies to the specified version _unless_ the cpanfile or other dependency overwrote the lock with something higher.

Most likely that should be a separate issue, blocked by this one, but not sure how to do that in Github, so we get a comment.
